Dockerfile

```diff
@@ -12,8 +12,7 @@ COPY ocrd_tesserocr ./ocrd_tesserocr
 COPY Makefile .
 RUN apt-get update && \
     apt-get install -y software-properties-common && \
-    add-apt-repository ppa:alex-p/tesseract-ocr && \
-    apt-get update && \
+    add-apt-repository -y ppa:alex-p/tesseract-ocr && \
     apt-get -y install --no-install-recommends \
     libtesseract-dev \
     tesseract-ocr \
```

Comment:
We should now also delegate to `deps-ubuntu` in the Dockerfile, of course.

Reply:
I am not convinced that this is the right solution. It would be wrong on hosts running very recent versions of Ubuntu or Debian and for people running their own version of Tesseract. How should those cases be handled?

For Docker and CircleCI the situation is different as we have a well defined environment. Perhaps the Makefile rule can be restricted to Ubuntu 18.04?

Reply:
> I am not convinced that this is the right solution. It would be wrong on hosts running very recent versions of Ubuntu or Debian and for people running their own version of Tesseract. How should those cases be handled?

I assume with `this` you refer to adding the PPA under `deps-ubuntu` in the Makefile.

Where else would you encapsulate system dependencies? This is precisely the convention that was introduced for OCR-D processors, and specifically for Ubuntu 18.04 to keep this manageable. ocrd_tesserocr should continue leading by good example.

As for running your own Tesseract build: `deps-ubuntu` does not prevent that – it does not even interfere (as long as you don't overwrite standard package directories).

Reply:
Adding a PPA is very special, and I really would feel better if that part were only done for Ubuntu 18.04.

In addition we could look for `/usr/local/include/tesseract` (or some other indicator that there is a local installation). If there is already such an installation, we should not add a 2nd installation as that can cause much confusion.

Reply:
> Adding a PPA is very special, and I really would feel better if that part were only done for Ubuntu 18.04.

But I just said `deps-ubuntu` denotes Ubuntu 18.04 under the current spec. 

You don't have to use that target if you don't like it, but it must be available when system dependencies exist, and must work automatically.

> In addition we could look for `/usr/local/include/tesseract` (or some other indicator that there is a local installation). If there is already such an installation, we should not add a 2nd installation as that can cause much confusion.

As I just said, `deps-ubuntu` does not interfere with local installation.  If the user wants that in addition, let her (keep that up-to-date). __Not doing__ system installation because of such heuristics (version checks??) would cause confusion.

Reply:
Now I am confused. You said that `deps-ubuntu` denotes Ubuntu 18.04, but that not doing system installation because of such heuristics (version checks??) would cause confusion?

To be more explicit: I see no problem with `deps-ubuntu` as long as the documentation clearly says what it is intended for (and in the case of `ocrd_tesserocr` when it should not be used) and as long as it is never called unexpectedly. It is ok when it is triggered by a user running `sudo make deps-ubuntu` or in the root context of a Docker build or CircleCI.

When I use `ocrd_all` and have installed Tesseract using `make install-tesseract`, I clearly don't want a 2nd installation.

Reply:
> Now I am confused. You said that `deps-ubuntu` denotes Ubuntu 18.04, but that not doing system installation because of such heuristics (version checks??) would cause confusion?

Yes, _by convention_! Not doing system-wide installation because of some checks in `/usr/local` would make that rule hard to predict and use.

> To be more explicit: I see no problem with `deps-ubuntu` as long as the documentation clearly says what it is intended for (and in the case of `ocrd_tesserocr` when it should not be used) and as long as it is never called unexpectedly. It is ok when it is triggered by a user running `sudo make deps-ubuntu` or in the root context of a Docker build or CircleCI.

But then we are in perfect agreement!

> When I use `ocrd_all` and have installed Tesseract using `make install-tesseract`, I clearly don't want a 2nd installation.

Then this is an issue specific to ocrd_all, it is off topic here. (You'd still want ocrd_tesserocr to encapsulate all of Tesseract's system dependencies though, if not Tesseract itself. Otherwise you end up copying recipes again.)